public/js/actions/tests/newSource.js

```diff
@@ -1,14 +1,21 @@
 "use strict";
 
-const { actions, selectors, createStore } = require("../../util/test-head");
+const { constants, selectors, createStore } = require("../../util/test-head");
 const fixtures = require("../../test/fixtures/foobar.json");
-const { newSource } = actions;
 const { getSourceByActor } = selectors;
 const sourcesFixtures = fixtures.sources.sources;
 
 const store = createStore();
 const expect = require("expect.js");
 
+// Write our own `newSource` to bypass the batching logic.
+function newSource(source) {
```

Comment:
I don't love this, but I'm not sure what else to do. Delaying things makes it hard to tests. One of the things I like about redux is it's so easy to think about the order in which things happen...

I really like the idea of implementing the throttling at the component level, because it's the sources component specifically that we want to avoid rerendering. That would solve this issue. More to think about...